derive_builder_core/src/builder_field.rs

```diff
@@ -76,6 +76,18 @@ impl<'a> BuilderField<'a> {
         let ident = self.field_ident;
         quote! { #ident : ::derive_builder::export::core::default::Default::default(), }
     }
+
+    /// Emits a builder field initializer that initializes the field to `Some(source.field)` or
+    /// `Default::default` in the case the field is not enabled.
+    /// Used to turn an instance of `Struct` back into a builder.
+    pub fn to_builder_initializer_tokens(&self) -> TokenStream {
+        if self.field_enabled {
```

Comment:
> What is the expected behavior if someone has provided custom setters?

I was not fully aware of custom setters when making these changes, been reading up on it and made some tests to verify my assumptions:

In my opinion it should not invoke the setter when converting an struct into a builder, the value should be moved, as is, from one to the other. This is because the contract of the custom setter is not known, e.g. it can expect *n* number of arguments of different types to produce the final value.

Re-setting/changing the value via the builder requires usage of the custom setter though.

Does this make sense? Have I missed something?

---

However, regarding the highlighted code where it emits the default constructor for non-enabled fields is problematic..!

Fields with `setter(skip)` and a default value/expr will be represented as `PhantomData<T>` in the builder, which prohibits setting the value in the `impl From` (if I'm not mistaken?).
The result is that these fields will regress back their default value, which of course would be very confusing if such a field had been updated during the lifetime of the struct!

Possible ways forward:

1. Define non-settable fields as `Option<T>` instead of `PhantomData<T>` and change the `build_fn` to use the default value only if the internal builder field is `None`. I encountered a test failure with this approach though in the [derive_trait](https://github.com/colin-kiegel/rust-derive-builder/blob/f58ecc719db790b4957734b6c6c7ebda30968789/derive_builder/tests/derive_trait.rs#L12-L15) test, the field type does not derive the expected traits (it didn't need to before). This is not well documented though but I assume there's a reason for the test. 

2. Document the corner case described above.

3. Hold off and regroup.


Wdyt?

Reply:
Personally, I need to take option 3 for a few weeks - both work and my personal life have gotten really busy and I can't commit the uninterrupted time needed to work through some scenarios for each case. Having some examples in Java where this is really used would be helpful, since it would provide insight into caller expectations.